modules/programs/fish.nix

```diff
@@ -45,109 +21,163 @@ in
-        type = types.attrsOf (types.either types.str (types.listOf types.str));
-        default = {};
+      vendor.completions.enable = mkOption {
+        type = types.bool;
+        default = true;
         description = ''
-          A set of environment variables used in the global environment.
-          These variables will be set on shell initialisation.
-          The value of each variable can be either a string or a list of
-          strings.  The latter is concatenated, interspersed with colon
-          characters.
+          Whether fish should use completion files provided by other packages.
+        '';
+      };
+      
+      vendor.functions.enable = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether fish should autoload fish functions provided by other packages.
         '';
-        apply = mapAttrs (n: v: if isList v then concatStringsSep ":" v else v);
       };
 
       shellAliases = mkOption {
-        type = types.attrs;
-        default = cfge.shellAliases;
+        default = config.environment.shellAliases;
         description = ''
           Set of aliases for fish shell. See <option>environment.shellAliases</option>
           for an option format description.
         '';
+        type = types.attrs;
       };
 
       shellInit = mkOption {
-        type = types.lines;
         default = "";
         description = ''
-          Shell Script code called during fish shell initialisation.
+          Shell script code called during fish shell initialisation.
         '';
+        type = types.lines;
       };
 
       loginShellInit = mkOption {
-        type = types.lines;
         default = "";
         description = ''
           Shell script code called during fish login shell initialisation.
         '';
+        type = types.lines;
       };
 
       interactiveShellInit = mkOption {
-        type = types.lines;
         default = "";
         description = ''
           Shell script code called during interactive fish shell initialisation.
         '';
+        type = types.lines;
       };
 
       promptInit = mkOption {
-        type = types.lines;
         default = "";
         description = ''
           Shell script code used to initialise fish prompt.
         '';
+        type = types.lines;
       };
 
     };
+
   };
 
   config = mkIf cfg.enable {
 
-    environment.systemPackages = [ pkgs.fish ];
-
-    environment.pathsToLink = [ "/share/fish" ];
-
-    environment.loginShell = "${shell}/bin/fish -l";
-    environment.variables.SHELL = "${shell}/bin/fish";
+    environment.etc."fish/foreign-env/shellInit".text = cfge.shellInit;
+    environment.etc."fish/foreign-env/loginShellInit".text = cfge.loginShellInit;
+    environment.etc."fish/foreign-env/interactiveShellInit".text = cfge.interactiveShellInit;
+    environment.etc."fish/foreign-env/extraInit".text= cfge.extraInit;
+
+    environment.etc."fish/nixos-env-preinit.fish".text = ''
```

Comment:
In the Fish package in Nixpkgs, Fish looks for this file, `/etc/fish/nixos-env-preinit.fish`. This is fine for us, but it means that we have to 'own' `/etc/fish` in `nix-darwin`. If we changed the Fish package to refer to `/etc/static/fish/nixos-env-preinit.fish` or something like that, this piece (and other pieces, if we changed them accordingly) would work even if `nix-darwin` can't or doesn't write directly to `/etc/fish`. Is that desirable? I don't think it would disrupt the way things work on NixOS.

tl;dr: Can we safely assume that `/etc/static` is where both nix-darwin and NixOS will write Fish config files to, and if so, should we change the modules (on both) to refer to those paths instead of the ones that live under `/etc`?